tests/test_release_notes.py

```
@@ -0,0 +1,199 @@
+def test_extract_changelog_section_matches_a_dated_heading_too() -> None:
+    """A heading with a real date, not just 'Unreleased', still matches."""
+    module = _module()
+    dated = _CHANGELOG.replace("## [0.2.0] - Unreleased", "## [0.2.0] - 2026-09-02")
+    section = module.extract_changelog_section(dated, "0.2.0")
+    assert "New thing." in section
+
+
+def test_extract_changelog_section_rejects_a_missing_version() -> None:
+    """A version with no CHANGELOG section must fail closed, not publish empty notes."""
+    module = _module()
+    with pytest.raises(ValueError, match="0.9.9"):
```

Ruff (0.16.3)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)